src/raw/aws/fargate_fetcher.go

```diff
@@ -88,12 +87,12 @@ func (e *FargateFetcher) fargateStatsFromCacheOrNew(response *FargateStats) erro
 // Note that the endpoint doesn't follow strictly the same schema as Docker's: it returns a list of containers,
 // instead of only one. They are not compatible in terms of the requests that they accept, but they share
 // part of the response's schema.
-func (e *FargateFetcher) getFargateContainerMetrics(stats *FargateStats) error {
+func (e *FargateFetcher) getFargateContainerMetrics() (FargateStats, error) {
 	endpoint := TaskStatsEndpoint(e.baseURL.String())
 
 	response, err := metadataResponse(e.http, endpoint)
```

Comment:
Shall we open an new issue to do a refactor here and make the code testable (in a way we can mock API response and easily reproduce this kind of problems)?

Reply:
Done. #136

Resolving :)